src/api/comms.js

```diff
@@ -80,3 +88,45 @@ export function deleteRequest(uri) {
     headers: getHeaders()
   });
 }
+
+export function generateBodyForSecretPatching(secretName) {
+  const patchAddBody = [
+    {
+      op: 'add',
+      path: 'serviceaccount/secrets/-',
+      value: {
+        name: secretName
+      }
+    }
+  ];
+
+  return patchAddBody;
+}
+
+export function patchRemoveSecretBody(indexOfSecret) {
+  const patchRemoveBody = [
+    {
+      op: 'remove',
+      path: `serviceaccount/secrets/${indexOfSecret}`
+    }
+  ];
+  return patchRemoveBody;
+}
+
+export async function patchAddSecret(uri, secretName) {
+  const patchAddBody = await generateBodyForSecretPatching(secretName);
```

Comment:
`generateBodyForSecretPatching` is a synchronous function, no need for `await` here
same with `patchRemoveSecretBody` and `getPatchHeaders` below

once these are updated you'll also no longer need the `async` keyword on the function definitions